core/src/main/java/org/elasticsearch/search/query/QueryPhase.java

```diff
@@ -138,7 +142,7 @@ public TopDocs call() throws Exception {
                         return new TopDocs(totalHitCountCollector.getTotalHits(), Lucene.EMPTY_SCORE_DOCS, 0);
                     }
                 };
-            } else if (searchContext.searchType() == SearchType.SCAN) {
+            } else if (searchType == SearchType.SCAN) {
```

Comment:
this confused me a bit, but it is here for bwc reasons. I do wonder if we can somehow use the `MinDocQuery` in such a way so the behaviour remains the same with scan search?